Content.Server/StationEvents/Events/RandomSpawnRule.cs

```diff
@@ -1,19 +1,32 @@
 using Content.Server.GameTicking.Rules.Components;
 using Content.Server.StationEvents.Components;
 using Content.Shared.GameTicking.Components;
+using Robust.Shared.Prototypes;
+using System.Linq;
 
 namespace Content.Server.StationEvents.Events;
 
 public sealed class RandomSpawnRule : StationEventSystem<RandomSpawnRuleComponent>
 {
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+
     protected override void Started(EntityUid uid, RandomSpawnRuleComponent comp, GameRuleComponent gameRule, GameRuleStartedEvent args)
     {
         base.Started(uid, comp, gameRule, args);
 
-        if (TryFindRandomTile(out _, out _, out _, out var coords))
+        for (var i = 0; i < 10; i++) // infinite loop prevention
         {
```

Comment:
Please don't do arbitrary loop checks unless the actual solution is unbounded (e.g. pathfinding).

Reply:
Have a better idea? try 10 is already what TryFindRandomTile does.

Reply:
I could stick this check into TryFindRandomTile, which I did think about, but it felt inelegant when that method gets used for other non-mob stuff a lot.

Reply:
The only truly bounded way I can think to do this is enumerating all the entities with the prevention component, getting their coords, and then outerjoining that with a dict of all tiles or something.

Reply:
> The only truly bounded way I can think to do this is enumerating all the entities with the prevention component, getting their coords, and then outerjoining that with a dict of all tiles or something.

I sat with this for a few hours looking at how to do it. I could probably do it with a snowflake method that only works for this specific component. Doing it in a more generic method for other components would mean I have to do something batty like reflection over EntityQueryEnumerator which I'm going to go out on a limb and say is a non-starter.